crates/workspace/src/pane.rs

```diff
@@ -1409,29 +1422,59 @@ impl Pane {
         )
     }
 
-    pub fn close_items_over_max_tabs(&mut self, window: &mut Window, cx: &mut Context<Self>) {
-        let Some(max_tabs) = WorkspaceSettings::get_global(cx).max_tabs.map(|i| i.get()) else {
```

Comment:
In the original code, we always pulled from the settings, but now that we are caching the `max_tabs` value on the `pane` struct, in the updated code, I'm just pulling from that instead inside of `close_tabs_on_file_open` and `close_tabs_on_settings_change`.